akka-cluster-bootstrap/src/main/scala/com/lightbend/rp/akkaclusterbootstrap/BootstrapExtension.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017 Lightbend, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.lightbend.rp.akkaclusterbootstrap
+
+import akka.actor._
+import akka.cluster.Cluster
+import akka.http.scaladsl.Http
+import com.lightbend.rp.servicediscovery.scaladsl._
+
+final class BootstrapExtensionImpl(system: ExtendedActorSystem) extends Extension {
+  val cluster = Cluster(system)
+  val http = Http(system)
+  val serviceLocator = ServiceLocator(system)
+}
+
+object BootstrapExtension extends ExtensionId[BootstrapExtensionImpl] with ExtensionIdProvider {
```

Comment:
Is there a reason why `BoostrapExtension` is not created using `ExtensionKey`? i.e.

```
object BootstrapExtension extends ExtensionKey[BootstrapExtensionImpl]
```

Doing it this way will be consistent with `service-discovery`'s `Settings` extension:

https://github.com/lightbend/reactive-lib/pull/3/files#diff-212517aab93835de83aab02b9e46218fR23

Reply:
I'm using the recommended way per Akka 2.5 docs: https://doc.akka.io/docs/akka/2.5/scala/extending-akka.html

I've updated the PR to make the usages consistent, changing `Settings` instead of `BootstrapExtension`.